### PR TITLE
KM473 🐛 Support build identifiers in tag

### PR DESCRIPTION
**Pull request by yngvark**
Converts tags to upgrader version, when using build identifiers in the tag.

## Description

`okctl upgrade` functionlity expects upgrade version to be on the form `0.0.1.something`. The upgrade version is read directly from the git tag. However, the `0.0.1.something` is an invalid semver *, and since GoReleaser expects tags to be valid semvers, we can't use the upgrade version as a tag.

This PR converts updates the release github action to convert the tag to an upgrade version, which is then used when making releases with GoReleaser.

So the new flow is:
* we make a new ugprade, and push it with the tag `0.0.1+somecomponent`
* github release action converts it to upgrade version: `0.0.1.somecomponent`, which is used by GoReleaser for naming binaries.

\* See `build identifier` in [valid semvers](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions)).

## Motivation and Context

Get upgrades to work with build identifiers in tag.

[KM473](https://trello.com/c/RJ1ubWLc/473-releasing-upgrades-fails-due-to-invalid-semantic-version)

## How to prove the effect of this PR?

When merged, it will be possible to tag commit on the form `0.0.1+something`, which will result in a release. (It's possible to edit the github action to test it for a branch as well.)

## Additional info

I have tested changes in https://github.com/oslokommune/okctl-upgrade-test.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -28,14 +28,3 @@
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-
-## Checklist:
```
Comment (fredriv) [inline .github/pull_request_template.md:32]: Are we removing the PR checklist?

Comment (yngvark) [reply .github/pull_request_template.md:32]: For this upgrade repository, I feel it doesn't make any sense, except maybe the first point.

Comment (fredriv) [reply .github/pull_request_template.md:32]: Ah, never mind I wasn't thinking about this being a different repo 😅

**Review by fredriv — APPROVED** [review]
LGTM 👍

**Review by deifyed — APPROVED** [review]

**Pull request merged**